website/blog/2024/08-20-problems-in-web-scraping/index.md

```diff
@@ -0,0 +1,368 @@
+
+Let's start with its advantages:
+
+- Allows you to bypass any Claudflare protection as well as many others using `botasaurus-driver`.
+- Good documentation for a quick start
+
+Downsides include:
+
+- Browser automation only, not intended for HTTP clients.
+- Tight coupling with `botasaurus-driver`; you can't easily replace it with something better if it comes out in the future.
+- No asynchrony, only multithreading.
+- At the moment, it's quite raw and still requires fixes for stable operation.
+- There are very few training materials available at the moment.
+
+This is a good framework for quickly building a web scraper based on browser automation. It lacks flexibility and support for HTTP clients, which is crutias for users like me.
+
+### [Crawlee for Python](https://www.crawlee.dev/python/)
+
+A new framework for web scraping in the Python ecosystem. The initial commit was made on Jan 10, 2024, with a release in the media space on July 5, 2024.
+
+:::tip
+If you like the blog so far, please consider [giving Crawlee a star on GitHub](https://github.com/apify/crawlee), it helps us to reach and help more developers.
+:::
+
+Developed by [Apify](https://apify.com/), it is a Python adaptation of their famous JS framework [`crawlee`](https://github.com/apify/crawlee), first released on Jul 9, 2019.
+
+As this is a completely new solution on the market, it is now in an active design and development stage. The community is also actively involved in its development. So,we can see that the use of [curl_cffi](https://github.com/apify/crawlee-python/issues/292) is already being discussed. The possibility of creating their own Rust-based client was [previously discussed](https://github.com/apify/crawlee-python/issues/80). I hope the company doesn't abandon the idea.
+
+:::note Crawlee team
+"Yeah, for sure we will keep improving Crawlee for Python for years to come."
+:::
+
+As I personally would like to see an HTTP client for Python developed and maintained by a major company. And Rust shows itself very well as a library language for Python. Let's remember at least [`Ruff`](https://docs.astral.sh/ruff/) and [`Pydantic`](https://docs.pydantic.dev/latest/) v2.
+
+Advantages:
+
+The framework was developed by an established company in the web scraping market, which has well-developed expertise in this sphere.
+- Support for both browser automation and HTTP clients.
+- Fully asynchronous, based on `asyncio`.
+- Active development phase and media activity. As developers listen to the community, it is quite important in this phase.
+
+On a separate note, it has a pretty good modular architecture. If developers introduce the ability to switch between several HTTP clients, we will get a rather flexible framework that allows us to easily change the technologies used, with a simple implementation from the development team.
```

Comment:
It is already possible to switch HTTP clients. But it's probably fair to keep the blog this way since it's not documented and we will probably change the API soon :smile: